src/docs/commands_keybindings.md

```diff
@@ -1,122 +1,142 @@
 ---
-title: Commands and Keybindings
+title: Commands/Menus/Keybindings
 ---
 
-# Commands and Keybindings
+# Commands, Menus and Keybindings
 
-Theia can be extended in a number of different ways. Commands allow packages to contribute unique commands that can then be called by other packages. It's also possible to add keybindings and contexts to these commands, so that they can only be called under certain conditions (window focus, current selection etc.).
+Commands are runnable actions defined by an ID and the function to be executed (plus some optional parameters like a name or an icon). Commands can be triggered via the command palette, they can be bound to keybindings or menu items and they can be called programmatically. The action that commands trigger can be context sensitive, so that they can only be called under certain conditions (window focus, current selection etc.).
 
-## Adding commands to Theia command registry
+The following sections provide details about how to contribute commands, keybindings and menu items. The sections will describe how to connect the different contributions and how to use the corresponding services for managing these items.
 
-To contribute commands to the command registry, modules must implement the `CommandContribution` class, i.e
+If you are not yet familiar with contribution points in Theia or the use of dependency injection, please consider this guide on [Services and Contributions](https://theia-ide.org/docs/services_and_contributions/).
 
-**java-commands.ts**
+All the following code examples are from the [Theia extension generator](https://github.com/eclipse-theia/generator-theia-extension). You can get the same code set-up by installing the generator, selecting the “Hello World” example (see here) and choosing “helloworld” as the name. 
```

Comment:
@JonasHelming the following is an introduction on `commands`, `menus` and `keybindings` but only describes `commands`. We should either describe each contribution type briefly, or move the information specific to commands in the `contributing commands` section:

> Commands are runnable actions defined by an ID and the function to be executed (plus some optional parameters like a name or an icon). Commands can be triggered via the command palette, they can be bound to keybindings or menu items and they can be called programmatically. The action that commands trigger can be context sensitive, so that they can only be called under certain conditions (window focus, current selection etc.).

Reply:
Sorry, I do not get this comment. It describes that "Commands can be triggered via the command palette, they can be bound to **keybindings** or **menu items**.
And then: "The following sections provide details about how to contribute commands, keybindings and menu items"
I think the command needs one sentence more, because it is the central concept but also not self-explanatory. If you know what a command is, I think you can understand what a menu item is that triggers this command.

Reply:
@JonasHelming we are targeting mainly newcomers to the project with the docs, I  was just thinking we should be as thorough as possible. At the moment we only describe commands at a high-level and not menus or keybindings.

In addition, not all menu items have commands (submenus, top-level menus), and keybindings can be expanded upon as well to refer to contexts or when clauses (these aren't trivial to newcomers).

Reply:
We can improve the docs in the future to cover additional use-cases that I mentioned, I just found it odd we described one of the contributions but not the others.

Reply:
> @JonasHelming we are targeting mainly newcomers to the project with the docs, I was just thinking we should be as thorough as possible. At the moment we only describe commands at a high-level and not menus or keybindings.

Yes, we are targeting newcomers. I think it is very important to not overload them with information in the intro. If you have to understand 10 things in one paragraph, I do not believe it is a good start. 
However, I gave it a try, but tbh, it is just weird:
 "A keybinding is a key combination triggering  a command"
 "A menu item is an item shown in a menu which can trigger a command". 
I think that just adds obvious information and makes the intro unnecessarily long.
Again, we do mention the concepts, I just think everybody knows what they mean. I believe I was never asked in a training what a menu item or a keybinding is, but I was asked what a command is.
Maybe it helps if we make the three concepts bold in the text?
I am sorry, but otherwise I am not really sure to address your comment, my attempts do not improve the doc for my PoV.  If you have a good suggestion on how to achieve what you want to say, please feel free to make a suggestion.

> 
> In addition, not all menu items have commands (submenus, top-level menus), and keybindings can be expanded upon as well to refer to contexts or when clauses (these aren't trivial to newcomers).

Keybinding contexts are deprecated. The when clause is described below, I do not think it should be mentioned in the intro, though. The hint on how to create sub menues and top level menus was indeed missing, good catch! I added that.